packages/apollo-server-core/src/runHttpQuery.ts

```diff
@@ -131,16 +133,12 @@ export async function runHttpQuery(
     // debug. Therefore, we need to do some unnatural things, such
     // as use NODE_ENV to determine the debug settings
     return throwHttpGraphQLError(500, [e as Error], {
-      debug: debugFromNodeEnv(process.env.NODE_ENV),
+      debug: debugFromNodeEnv(NODE_ENV),
```

Comment:
Probably at least extend the comment above to note that unlike other aspects of NODE_ENV in this package, this one isn't sensitive to runtime changes to the environment variable.

(Alternatively, perhaps it's OK performance-wise to read the environment variable in this 500 situation?)

Reply:
> Alternatively, perhaps it's OK performance-wise to read the environment variable in this 500 situation

Yeah, I suppose this isn't a hot code path. I imagine an error here would be something the would only happen intermittently... and if it's happening more than that, you likely got bigger problems than a runtime cost!

Either way, in my new revision I have `debugFromNodeEnv` default to the "cached" `NODE_ENV`, so I added the comment you requested. I think that this keeps it consistent with the default behaviour (ie. `nodeEnv` defaults to `NODE_ENV` and is also not sensitive to runtime changes).